### ISSUE TITLE
--verbose-errors should cause entire rows to be shown

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#3722]
For example:

```purescript
module Main where

foo = { a: 1, b: 2, c: 3 }
bar = { a: 1, b: 2, c: "3" }
baz = [foo, bar]
```

When compiling with `--verbose-errors` we get:
```
Error found:
in module Main
at src/Main.purs:5:1 - 5:17 (line 5, column 1 - line 5, column 17)
at src/Main.purs:5:7 - 5:17 (line 5, column 7 - line 5, column 17)

  Could not match type
       
    Int
       
  with type
          
    String
          

while trying to match type Int
  with type String
while trying to match type             
                             ( c :: Int
                             ...       
                             )         
                                       
  with type                
              ( c :: String
              ...          
              )            
                           
while trying to match type t0
  with type { a :: Int   
            , b :: Int   
            , c :: String
            }            
while inferring the type of [ foo
                            , bar
                            ]    
in value declaration baz

where t0 is an unknown type

See https://github.com/purescript/documentation/blob/master/errors/TypesDoNotUnify.md for more information,
or to contribute content related to this error.
```

But I think we should show the entire row type if the `--verbose-errors` flag is specified.

**Comment by Woody88** [comment]
Hi. 

I was trying to have a look at this but I am getting a different output.  
```
Warning 1 of 2:

  at Main.hs:3:1 - 3:27 (line 3, column 1 - line 3, column 27)
  at Main.hs:3:1 - 3:27 (line 3, column 1 - line 3, column 27)

    No type declaration was provided for the top-level declaration of foo.
    It is good practice to provide type declarations as a form of documentation.
    The inferred type of foo was:
                
      { a :: Int
      , b :: Int
      , c :: Int
      }         
                

  in value declaration foo

  See https://github.com/purescript/documentation/blob/master/errors/MissingTypeDeclaration.md for more information,
  or to contribute content related to this warning.

Warning 2 of 2:

  at Main.hs:4:1 - 4:29 (line 4, column 1 - line 4, column 29)
  at Main.hs:4:1 - 4:29 (line 4, column 1 - line 4, column 29)

    No type declaration was provided for the top-level declaration of bar.
    It is good practice to provide type declarations as a form of documentation.
    The inferred type of bar was:
                   
      { a :: Int   
      , b :: Int   
      , c :: String
      }            
                   

  in value declaration bar

  See https://github.com/purescript/documentation/blob/master/errors/MissingTypeDeclaration.md for more information,
  or to contribute content related to this warning.


Error found:

  Unable to write file:

    output/cache-db.json


```
Steps taken:
- I cloned this repo
- executed `stack build`
- executed `stack test` 
- created a Main.hs in the root and pasted the code above in it.
- executed `stack run -- compile Main.hs --verbose-errors`

when I check the version of the compiler `stack run -- --version` I get this output:
```0.13.3 [development build; commit: 4ba1b2a6b4451aeb0908926898328cc346dece06 DIRTY]```

Can you please tell me what I'm doing wrong?

Thank you,

Woodson D.

**Comment by hdgarrood** [comment]
I think you might have run into a separate bug here. You might be able to work around it by creating the `output` directory manually.

**Comment by hdgarrood** [comment]
That might be a bug I introduced during #3705

**Comment by hdgarrood** [comment]
Yes, this is definitely a new bug. I'll open a new issue. Thanks for the report!

**Comment by hdgarrood** [comment]
@Woody88 should be fixed now (see #3729). Can you pull the latest `master` and try again?

**Comment by Woody88** [comment]
@hdgarrood  Wow, so quick! :) I was going to ask you if you wanted me to create an issue for it, and I was also going to let you know that I found which file might be causing the issue but you already fixed everything! haha. 

>  Can you pull the latest master and try again?
I did and it works now! Thank you very much.

This week is a long weekend so I should be able to fix this issue by tomorrow.
